### ISSUE TITLE
tf2caffe, resnet_v2_152, got NAN while testing caffe model

### ISSUE THREAD

**Issue by runninging** [microsoft/MMdnn#690]
I convert the tf model according to the official steps [here](https://github.com/Microsoft/MMdnn/blob/master/mmdnn/conversion/tensorflow/README.md) when i run the test in caffe, i got nan as follows:

{'MMdnn_Output': array([nan, nan, nan, ..., nan, nan, nan], dtype=float32)}

![屏幕快照 2019-06-26 下午6 19 48](https://user-images.githubusercontent.com/30371890/60172433-3d47fb80-983f-11e9-93ba-5b7c0e5ee72b.png)

Could anyone help me with that? Thanks a lot!

**Comment by Guodonggogo** [comment]
In my case, NaN resulted from batchnormalization layers and  it is because the variances in batchnormalization layer and the scales in bn_scale layer are overwritten. I modified function "rename_BatchNorm" in mxnet_parser.py as:
    def rename_BatchNorm(self, source_node):
        IR_node = self._convert_identity_operation(source_node)

        # axis
        if self.data_format in MXNetParser.channels_first or self.data_format == 'None':
            IR_node.attr["axis"].i = MXNetParser._convert_axis(IR_node, int(source_node.get_attr("axis", "1")))
        else:
            IR_node.attr["axis"].i = int(source_node.get_attr("axis", "1"))

        # scale
        IR_node.attr["scale"].b = not MXNetParser.str2bool(source_node.get_attr("fix_gamma", "True"))
        IR_node.attr["bias"].b = True
        # epsilon
        IR_node.attr["epsilon"].f = float(source_node.get_attr("eps", "0.001"))

        # momentum
        IR_node.attr["momentum"].f = float(source_node.get_attr("momentum", "0.9"))

        # weights
        if self.weight_loaded:
            source_node_name = source_node.name.replace('_fwd','')
            # gamma
            if IR_node.attr["scale"].b:
                self.set_weight(source_node.name, "gamma_bn", self.weight_data.get(source_node_name + "_gamma").asnumpy())

            # beta
            if IR_node.attr["bias"].b:
                self.set_weight(source_node.name, "beta", self.weight_data.get(source_node_name + "_beta").asnumpy())

            # mean
            try:
               self.set_weight(source_node.name, "mean", self.weight_data.get(source_node_name + "_moving_mean").asnumpy())
            except:
               self.set_weight(source_node.name, "mean", self.weight_data.get(source_node_name + "_running_mean").asnumpy())

            # var
            try:
               self.set_weight(source_node.name, "var", self.weight_data.get(source_node_name + "_moving_var").asnumpy())
            except:
               self.set_weight(source_node.name, "var", self.weight_data.get(source_node_name + "_running_var").asnumpy())

and the fuction "gen_weight" in 4d6fae9b7fc54ae7aa29268346551e6d.py (architecture file generated by mmdnn) as follows:
def gen_weight(weight_file, model, prototxt):
    global __weights_dict
    __weights_dict = load_weights(weight_file)

    net = caffe.Net(prototxt, caffe.TRAIN)

    for key in __weights_dict:
        if 'weights' in __weights_dict[key]:
            net.params[key][0].data.flat = __weights_dict[key]['weights']
        elif 'mean' in __weights_dict[key]:
            net.params[key][0].data.flat = __weights_dict[key]['mean']
            net.params[key][1].data.flat = __weights_dict[key]['var']
            if 'gamma_bn' in __weights_dict[key]:
                net.params[key+'_scale'][0].data.flat = __weights_dict[key]['gamma_bn']
                net.params[key+'_scale'][1].data.flat = __weights_dict[key]['beta']
        elif 'scale' in __weights_dict[key]:
            net.params[key][0].data.flat = __weights_dict[key]['scale']
        if 'bias' in __weights_dict[key]:
            net.params[key][1].data.flat = __weights_dict[key]['bias']
        if 'gamma' in __weights_dict[key]: # used for prelu, not sure if other layers use this too
            net.params[key][0].data.flat = __weights_dict[key]['gamma']
    net.save(model)
    return net
Then, it works.
